platform/posix/transport/src/openssl_posix.c

```diff
@@ -401,6 +402,7 @@ OpensslStatus_t Openssl_Connect( NetworkContext_t * pNetworkContext,
     OpensslStatus_t returnStatus = OPENSSL_SUCCESS;
     long verifyPeerCertStatus = X509_V_OK;
     int sslStatus = 0;
+    uint8_t sslObjectCreated = 0;
```

Comment:
Can we instead initialize the pNetworkContext->pSsl to NULL?

Reply:
We could do that to avoid `sslObjectCreated`, though I was trying not to change the value if not required. 
Thinking about it again, I am fine with setting `pNetworkContext->pSsl` to NULL instead of using the `sslObjectCreated` flag

Reply:
Actually, there is no clear way of setting the `pNetworkContext->pSsl` to NULL as the `pNetworkContext` can be an invalid input itself, in which case we will hit the ` if( ( returnStatus != OPENSSL_SUCCESS ) && .... ))` case again for cleaning up the SSL object. 

The alternative would be to check if the `returnStatus != OPENSSL_INVALID_PARAMETER &&  returnStatus != OPENSSL_SUCCESS` which I think is not expressive as `returnStatus != OPENSSL_SUCCESS && sslObjectCreated == 1u`, thus I prefer the latter

Reply:
Thanks for explaining to me, I understand now.

Reply:
Thank you for applying this fix